packages/loader/container-loader/src/container.ts

```diff
@@ -310,7 +316,9 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
                         container.removeListener("closed", onClosed);
                     })
                     .then((props) => {
-                        event.end({ ...props, ...loadOptions.loadMode });
+                        event.end({
+                            details: JSON.stringify({ ...props }),
```

Comment:
I personally find it easier to see all the props on a single event, instead of them being split in two places. So I'd recommend adding canReconnect here (if that's what's missing) and not change start event. Or do both, if we are concerned about load never finishing and us not getting this data.
Please note that with "regular" properties there is no problem with adding more properties on end event - all the props that were supplied for start event will show up on end event as well. But when we push all properties into "details", overwrite (not merge) happens, and end event does not have proper that exist on start event. I'd say - if we want to have props on both events, keep the old structure of logging individual properties without grouping them into "details"

Reply:
> But when we push all properties into "details", overwrite (not merge) happens

Yeah this has been annoying in other PRs too.  I'm thinking we should allow flat objects as telemetry props (maybe update #12119?) so callers can build up the object and then have it serialized at the end

Reply:
its a huge topic, but generally databases are really good a storing sparse columnar data with lots of repeated values. they are also good a partition skipping which allows skipping big chunks of data for efficient querying. These features are also frequently interrelated. given that we should be careful what we bundle, and what we don't bundle, as we could make storage and querying worse. Generally, if something only applies to a single log line, bundling is fine, but if you even want to query on it independently, or it is relevant across multiple logs line it should not be bundled.